### PR TITLE
ci(jenkins): Enable CI Pipeline

### PR DESCRIPTION
**Pull request by v1v**
### What

Enable CI Pipeline

### Why

Jenkins is the CI/CD tool we have in place.

### CODE REVIEW

File: .ci/scripts/lint.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+set -e
+
+echo 'TBD'
```
Comment (v1v) [inline .ci/scripts/lint.sh]: This is something to be done in a follow-up

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+repos:
```
Comment (v1v) [inline .pre-commit-config.yaml]: This is something to be done in a follow-up

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      }
+    }
+    stage('Test') {
+      options { skipDefaultCheckout() }
+      steps {
+        deleteDir()
+        unstash "source"
+        dir("${BASE_DIR}"){
+          script {
+            def mapParallelTasks = [:]
```
Comment (kuisathaverat) [inline .ci/Jenkinsfile]: since we have the matrix step I'd preferred to use it instead of each block, then you can remove the YAML file too

Comment (v1v) [reply .ci/Jenkinsfile]: Sounds good, I was tempted to, but I liked the way to define the supported python versions somewhere else. So I'll apply the matrix approach then

**Comment by v1v** [comment]
I'll merge it now as this is the skeleton for now

**Pull request merged**